ModulemdTranslationHelpers/Utils.py

```diff
@@ -58,3 +63,94 @@ def get_translation_catalog_from_index(index, project_name):
         catalog.add(translatable_string, locations=locations)
 
     return catalog
+
+
+def get_modulemd_translations_from_catalog(catalogs, index):
+    # Nested dictionary `data` contains information from catalog like:
+    # Key_1: <module>:<stream>
+    # Value_1: Key_2: <locale>
+    #          Value_2: list of (<string_type>:<msg.string>)
+
+    data = defaultdict(defaultdict)
+
+    for catalog in catalogs:
+        for msg in catalog:
+            for location, _ in msg.locations:
+                (module_name, stream_name, string_type,
+                 profile_name) = split_location(location)
+
+                key_1 = ("{}:{}").format(module_name, stream_name)
+                if profile_name is None:
+                    value_2 = ("{}:{}").format(string_type, msg.string)
+                else:
+                    value_2 = ("{}:{}:{}").format(
+                        string_type, profile_name, msg.string)
+
+                # Adding into data
+                data[key_1].setdefault(
+                    catalog.locale_identifier, []).append(value_2)
+
+    for mod_stream, translations in data.items():
+        (module_name, stream_name) = mod_stream.split(":")
+        now = datetime.utcnow()
+        modified = int("%04d%02d%02d%02d%02d%02d" % (
+            now.date().year,
+            now.date().month,
+            now.date().day,
+            now.time().hour,
+            now.time().minute,
+            now.time().second
+        ))
+        mmd_translation = Modulemd.Translation.new(
+            1, module_name, stream_name, modified)
+
+        for locale, strings in translations.items():
+            entry = Modulemd.TranslationEntry.new(locale)
+
+            for string in strings:
+                (string_type, profile_name,
+                 translation) = split_translation_string(string)
+
+                if string_type == "summary":
+                    entry.set_summary(translation)
+                elif string_type == "description":
+                    entry.set_description(translation)
+                else:
+                    entry.set_profile_description(profile_name, translation)
+
+            mmd_translation.set_translation_entry(entry)
+
+        ret = index.add_translation(mmd_translation)
+        if not ret:
+            print("Error!")
+
+    return None
+
+
+def split_translation_string(string):
```

Comment:
This entire function should go away, since we should just be operating on the TranslationEntry itself instead of duplicating effort by creating a list of joined strings only to have to split them apart again moments later.